openmole/plugins/org.openmole.plugin.tool.pattern/src/main/scala/org/openmole/plugin/tool/pattern/Strain.scala

```diff
@@ -28,7 +29,14 @@ object Strain {
     val first = Capsule(EmptyTask(), strainer = true)
     val last = Slot(Capsule(EmptyTask(), strainer = true))
 
-    (first -- puzzle -- last) + (first -- last)
+    val mole = puzzle.toMole
+    val sources = Sources(puzzle.sources.groupBy(_._1).mapValues(_.map(_._2)))
+    val hooks = Hooks(puzzle.hooks.groupBy(_._1).mapValues(_.map(_._2)))
```

Comment:
I tried to move the implicit conversion functions into the companion objects but then received a number of errors in the SubMoleExecution because it apparently didn't find the mapToHooks|Sources or vice versa implicit any more. Thus, I just added the new ones directly to the package object.